doc/cprover-manual/jbmc-user-manual.md

```diff
@@ -0,0 +1,234 @@
+```
+package tutorial;
+
+public class Example2
+    public static int query(int[] array, int index) {
+        if (array == null) {
+            return -1;
+        }
+        if (index < 0 || index >= array.length) {
+            return -1;
+        }
+        return array[index];
+    }
+
+}
+```
+then the JBMC automatic assertions become valid, meaning that there is no
+possible inputs (argument values) for which they can be violated:
+
+    [java::tutorial.Example2.query:([II)I.1] line 6 no uncaught exception: SUCCESS
+    [java::tutorial.Example2.query:([II)I.null-pointer-exception.1] line 9 Null pointer check: SUCCESS
+    [java::tutorial.Example2.query:([II)I.null-pointer-exception.2] line 12 Null pointer check: SUCCESS
+    [java::tutorial.Example2.query:([II)I.array-index-out-of-bounds-low.1] line 12 Array index should be >= 0: SUCCESS
+    [java::tutorial.Example2.query:([II)I.array-index-out-of-bounds-high.1] line 12 Array index should be < length: SUCCESS
+
+
+\subsection jbmc-manual-unwind Loop unwinding
+
+As CBMC, JBMC transforms the input program into a state representation,
+and thus needs to unwind loops up to a given value. Consider the `isPrime` 
+method in the code below. Without specifying an unwind limit, JBMC will not 
+terminate when analysing the following function whose state representation is
+theoretically unbounded (because of the for-loop whose number of iterations 
+depends on an input):
+```
+package tutorial;
+
+public class Example3 {
+
+    public static void doSomething(int inputVal) {
+        if (inputVal > 1 && inputVal % 2 == 1) {
+            assert isPrime(inputVal);
+            // do something
+        }
+    }
+
+    public static boolean isPrime(int candidate) {
+        for (int div = 2; div * div <= candidate; div++) {
+            if (candidate % div == 0) {
+                return false;
+            }
+        }
+        return candidate > 1;
+    }
+
+}
+ ```
+To limit the number of times the for-loop is unwound, we use the `--unwind N`
+options, in which case the following call to JBMC:
+
+    jbmc Example3.class --function tutorial.Example3.isPrime --unwind 10
+
+will terminate (with `VERIFICATION SUCCESSFUL`).
+
+
+\subsection jbmc-manual-user-assert User assertions
+
+Elaborating on the example above, users may provide their own assertions, that
+JBMC will try do refute. On line 7, we check the assertion that all odd 
+numbers greater than 1 are prime. To be sure that this always holds, we run 
+JBMC on the example, with a reasonable `unwind` value:
+
+    jbmc Example3.class --function tutorial.Example3.doSomething --unwind 10
+
+Unsurprisingly JBMC doesn't agree, and prints an assertion failure
+(truncated here for readability):
+
+    [...doSomething:(I)V.assertion.1] line 7 assertion at file tutorial/Example3.java: FAILURE
+
+
+Rerunning the analysis with the `--trace` option, we can see find out which
+input value JBMC found to trigger the violation:
+
+    INPUT inputVal: 15
```

Comment:
When I run `jbmc Example3.class --function tutorial.Example3.doSomething --unwind 10 --trace`, I did not see `INPUT inputVal: 15` but it shows me ```Violated property:
  file tutorial/Example3.java function tutorial.Example3.doSomething(int) line 6 thread 0
  assertion at file tutorial/Example3.java line 6 function java::tutorial.Example3.doSomething:(I)V bytecode-index 16
  false



** 1 of 15 failed (2 iterations)
VERIFICATION FAILED
```. Have I miss something there?

Reply:
Don't you see something like this in the trace view? I rephrased the doc a bit to suggest that it needs a bit of searching.